test/test_general.py

```diff
@@ -99,73 +98,86 @@ def debug_nondeterminism(b, c):
     mnist, labels = openml_datasets["mnist_784"]
     mnist = mnist.reshape(mnist.shape[0], -1)
-
-    # Use subset for faster testing
-    mnist = mnist[:1000]
-    labels = labels[:1000].astype(int)
+    labels = labels.astype(int)
 
     reducer = pacmap.PaCMAP(n_components=2, n_neighbors=10, MN_ratio=0.5, FP_ratio=2.0, random_state=20)
     embedding = reducer.fit_transform(mnist, init="pca")
-    test_utils.generate_figure(embedding, labels, 'test_mnist_seed')
+    test_utils.generate_figure(embedding, labels, 'test_mnist_full_seed')
+
+def test_pacmap_mnist_full_noseed(openml_datasets):
+    """Test PaCMAP with full MNIST dataset from OpenML without seed."""
+    # Load MNIST from fixture
+    mnist, labels = openml_datasets["mnist_784"]
+    mnist = mnist.reshape(mnist.shape[0], -1)
+    labels = labels.astype(int)
+
+    reducer = pacmap.PaCMAP(n_components=2, n_neighbors=10, MN_ratio=0.5, FP_ratio=2.0, save_tree=True)
+    embedding = reducer.fit_transform(mnist, init="pca")
+    test_utils.generate_figure(embedding, labels, 'test_mnist_full_noseed')
+
+def test_pacmap_mnist_subset_save_load(tmp_path, openml_datasets):
+    """Test PaCMAP save/load with subset MNIST dataset from OpenML."""
+    # Load MNIST from fixture
+    mnist, labels = openml_datasets["mnist_784"]
+    mnist = mnist.reshape(mnist.shape[0], -1)
 
-    plt.savefig("./test/output/test_mnist_seed.png")
+    # Use subset for faster testing
+    mnist = mnist[:1000]
+    labels = labels[:1000].astype(int)
 
     reducer = pacmap.PaCMAP(n_components=2, n_neighbors=10, MN_ratio=0.5, FP_ratio=2.0, save_tree=True)
     embedding = reducer.fit_transform(mnist, init="pca")
-    fig, ax = plt.subplots(1, 1, figsize=(6, 6))
-    ax.scatter(embedding[:, 0], embedding[:, 1], s=0.5, c=labels, cmap='Spectral')
-    ax.axis('off')
-    ax.set_title('test_mnist_noseed')
-    plt.savefig("./test/output/test_mnist_noseed.png")
 
-    # Save and load
+    # Save
     save_path = tmp_path / "mnist_reducer"
     pacmap.save(reducer, str(save_path))
+
     embedding = reducer.transform(mnist)
-    fig, ax = plt.subplots(1, 1, figsize=(6, 6))
-    ax.scatter(embedding[:, 0], embedding[:, 1], s=0.5, c=labels, cmap='Spectral')
-    ax.axis('off')
-    ax.set_title('test_saveload')
-    plt.savefig("./test/output/test_saveload_before.png")
+    test_utils.generate_figure(embedding, labels, 'test_saveload_subset_before')
 
+    # Load
     reducer = pacmap.load(str(save_path))
+
     embedding = reducer.transform(mnist)
-    fig, ax = plt.subplots(1, 1, figsize=(6, 6))
-    ax.scatter(embedding[:, 0], embedding[:, 1], s=0.5, c=labels, cmap='Spectral')
-    ax.axis('off')
-    ax.set_title('test_saveload')
-    plt.savefig("./test/output/test_saveload_after.png")
+    test_utils.generate_figure(embedding, labels, 'test_saveload_subset_after')
 
     print("Figures have been generated successfully.")
 
 # TODO: Remove this is we want to commit to using pytest.
 if __name__ == "__main__":
```

Comment:
Would love to just remove this from the end of each test file if we're happy with pytest as a test runner.

This essentially allows someone to run tests with `python test/test_general.py` instead of `pytest test/test_general.py`. It was added my me as a way to introduce pytest without breaking someone else's prior muscle memory for running the tests.

Happy to keep this PR simple as just merge as-is. Can remove this in another PR if you're in favour :)